specs/IFrameMemoryUsage.md

```diff
@@ -0,0 +1,306 @@
+IFrame Memory Usage API
+===
+
+# Background
+The WebView2 team has been asked to provide support for monitoring iframe 
+memory usage. The end developer needs to know how much memory each iframe 
+consumes. To do that, they need to get the renderer process ID for each 
+iframe. However, WebView2 only tracks the render frame host for top-level iframes. 
+This limits the [ICoreWebView2Frame2](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2frame?view=webview2-1.0.1518.46)
+APIs for first-level iframes only. For instance, we raise the FrameCreated 
+event for first-level iframes, but not for nested iframes with depth greater than 1. 
+Also, WebView2 lacks an API to get the renderer process ID for iframes.
+<pre>
+                           main_frame  [depth:0]
+                         /     |       \
+      first_level_iframe_0     ...    first_level_iframe_n             [depth:1, first level iframe]
+                |          \                     |
+      second_level_iframe_0 ...       second_level_iframe_n ...        [depth:2, second level iframe]
+               |
+              ...                                                      
+</pre> 
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2` and `CoreWebView2Frame` to include the 
+`RendererProcessId` property and  `RendererProcessIdChanged` event. This property 
+stores the renderer process ID for its iframe. This event will be raised whenever 
+the renderer process ID is changed for `CoreWebView2` or `CoreWebView2Frame`.
+
+Additionally, we propose to add the `FrameCreated` event to `CoreWebView2Frame`. 
+This event is raised when a new iframe is created.
+
+# Examples
+C++
+```c++
+void InitializeEventView(wil::com_ptr<ICoreWebView2> webview)
+{
+    auto webview2_18 = webview.try_query<ICoreWebView2_18>();
+    if (webview2_18) 
+    {
+        webview2_18->add_FrameCreated(
+            Callback<ICoreWebView2FrameCreatedEventHandler>(
+                [this](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args)
+                    -> HRESULT {
+                    // ...
+                    // The new frame created event can be recursively initialized
+                    InitializeFrameEventView(webviewFrame);

```

Comment:
Where did this magic `webviewFrame` variable come from?
```cpp
wil::com_ptr<ICoreWebView2Frame> webviewFrame;
CHECK_FAILURE(args->get_Frame(&webviewFrame));
```

Reply:
Please fix sample code.